hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/DefaultWriteBehindProcessor.java

```diff
@@ -360,13 +360,16 @@ public void callAfterStoreListeners(Collection<DelayedEntry> entries) {
         boolean result = false;
         Exception exception = null;
         int k = 0;
+        boolean interrupted = false;
         for (; k < RETRY_TIMES_OF_A_FAILED_STORE_OPERATION; k++) {
             try {
                 result = task.run();
+            } catch (InterruptedException ex) {
+                interrupted = true;
```

Comment:
I see you are breaking the loop; but why don't you do it as soon as you catch the exception? What is the advantage of introducing a variable and after the loop check this variable?

Apart from that; you are loosing the interrupt flag. This is a problem because you might have broken free from whatever happened in the run method; any subsequent blocking call won't get interrupted.